experiment/secret-sync-controller/client/client.go

```diff
@@ -0,0 +1,48 @@
+package client
+
+import (
+	"context"
+	"flag"
+	"os"
+	"path/filepath"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"google.golang.org/api/option"
+
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func NewK8sClientset() *kubernetes.Clientset {
```

Comment:
Consider using `kubernetes.Interface` instead

Reply:
meta comment for the function as a whole: consider re-using https://github.com/kubernetes/test-infra/blob/master/prow/kube/cluster.go#L57 instead (you are welcome to do this as a followup PR instead)

Reply:
It looks like https://github.com/kubernetes/test-infra/blob/19a078d68a69b0eaa948876f0e2b4f64fc9c78df/prow/kube/cluster.go#L57 tries to use in-cluster config first, and falls back to default rules if failed.
This would be optimal for me. Re-using this function.